source/slang/slang-check-decl.cpp

```diff
@@ -1443,6 +1443,325 @@ namespace Slang
         return false;
     }
 
+    bool SemanticsVisitor::trySynthesizeMethodRequirementWitness(
+        ConformanceCheckingContext* context,
+        LookupResult const&         lookupResult,
+        DeclRef<FuncDecl>           requiredMemberDeclRef,
+        RefPtr<WitnessTable>        witnessTable)
+    {
+        // The situation here is that the context of an inheritance
+        // declaration didn't provide an exact match for a required
+        // method. E.g.:
+        //
+        //      interface ICounter { [mutating] int increment(); }
+        //      struct MyCounter : ICounter
+        //      {
+        //          int increment(int val = 1) { ... }
```

Comment:
Is `[mutation]` needed for implementations?

Reply:
Oh yeah, it should be required there.

One note is that my current test case doesn't cover `static` or `[mutating]` methods, so I can't pretend to have full confidence that they will work (I'd like to expand out the test coverage bit by bit so that I can keep the PRs/fixes small).